src/middlewared/middlewared/plugins/filesystem.py

```diff
@@ -142,6 +142,9 @@ def listdir(self, path, filters, options):
         if not path.is_dir():
             raise CallError(f'Path {path} is not a directory', errno.ENOTDIR)
 
+        if 'ix-applications' in path.parts:
+            raise CallError('Ix-applications is a system managed dataset and it\'s contents cannot be listed')
```

Comment:
```suggestion
            raise CallError('Ix-applications is a system managed dataset and its contents cannot be listed')
```
`it's` is a contraction for `it is` or 'it has' so remove the apostrophe.